app/src/main/java/org/torproject/android/BasicAuthentication.java

```diff
@@ -0,0 +1,40 @@
+package org.torproject.android;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.charset.Charset;
+
+import cz.msebera.android.httpclient.Header;
+import cz.msebera.android.httpclient.HttpResponse;
+import cz.msebera.android.httpclient.auth.AuthenticationException;
+import cz.msebera.android.httpclient.auth.UsernamePasswordCredentials;
+import cz.msebera.android.httpclient.client.HttpClient;
+import cz.msebera.android.httpclient.client.methods.HttpGet;
+import cz.msebera.android.httpclient.impl.auth.BasicScheme;
+import cz.msebera.android.httpclient.impl.client.HttpClientBuilder;
+
+public class BasicAuthentication {
+    public void basicAuth(){
+        StringBuilder builder = new StringBuilder();
+        HttpClient client = HttpClientBuilder.create().build();
+        HttpGet request = new HttpGet("http://127.0.0.1:8080/");
```

Comment:
The onion address embedded here is really the key thing we'll want to be configurable.

Reply:
This doesn't appear to use the SOCKS5 proxy to reach a tor hidden service.

Reply:
That's some old testing code that can be removed. The actual code used is in `RequestTorConfigTask.java` and it uses an http proxy setup by orbot which can be used to connect to hidden services through http apis.